### PR TITLE
Revised animal excretion

### PR DESCRIPTION
**Pull request by TaranRallings**
# Description

This is a fairly straightforward change providing a revised metabolic framework. The pre-existing metabolize and excrete methods have been reconfigured into:
- metabolize: for the generating of excreta mass (metabolic wastes)
- excrete: for the removal of nitrogenous wastes into soil pools (urination and analogs)
- respire: for the removal of carbonaceous wastes into the atmosphere (breathing CO2)
- defecate: for the removal of digestive wastes into soil pools

As we are not currently tracking nitrogen or water, the excrete trait and method does nothing but provide structure.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [ ] Relevant documentation reviewed and updated

### CODE REVIEW

**Review by jacobcook1995 — COMMENTED** [review]
The excretion stuff all looks sensible to me! (I don't have any feedback on the code, but that's probably because I'm not familiar enough with the approach you are taking)

File: virtual_ecosystem/models/animal/animal_cohorts.py [hunk 1]
```diff
@@ -133,7 +127,41 @@ def metabolize(self, temperature: float, dt: timedelta64) -> float:
         # in data object
         return actual_mass_metabolized * self.individuals
 
-    def excrete(
+    def excrete(self, excreta_mass: float, excrement_pool: DecayPool) -> None:
+        """Transfers nitrogenous metabolic wastes to the excrement pool.
+
+        This method will not be fully implemented until the stoichiometric rework. All
+        current metabolic wastes are carbonaceous and so all this does is provide a link
+        joining metabolism to a soil pool for later use.
```
Comment (dalonsoa) [inline virtual_ecosystem/models/animal/animal_cohorts.py:135]: Should this have a `TODO` so it is easier to find and amend when relevant?

File: virtual_ecosystem/models/animal/animal_cohorts.py [hunk 2]
```diff
@@ -133,7 +127,41 @@ def metabolize(self, temperature: float, dt: timedelta64) -> float:
...
+            excreta_mass * self.constants.nitrogen_excreta_proportion
+        )
+
+    def respire(self, excreta_mass: float) -> float:
+        """Transfers carbonaceous metabolic wastes to the atmosphere.
+
+        This method will not be fully implemented until the stoichiometric rework. All
+        current metabolic wastes are carbonaceous and so all this does is return the
+        excreta mass for updating data["total_animal_respiration"] in metabolize
+        community.
```
Comment (dalonsoa) [inline virtual_ecosystem/models/animal/animal_cohorts.py:153]: And the same here.

File: virtual_ecosystem/models/animal/animal_cohorts.py [hunk 3]
```diff
@@ -697,34 +723,6 @@ def is_below_mass_threshold(self, mass_threshold: float) -> bool:
             self.mass_current + self.reproductive_mass
         ) / self.functional_group.adult_mass < mass_threshold
 
-    def inflict_natural_mortality(
```
Comment (dalonsoa) [inline virtual_ecosystem/models/animal/animal_cohorts.py:700]: This function is removed but not added anywhere else and, also, not related with the purpose of the PR. Is it an error or am I missing anything?

File: virtual_ecosystem/models/animal/scaling_functions.py [hunk 4]
```diff
@@ -36,35 +36,6 @@ def damuths_law(mass: float, terms: tuple) -> int:
     return ceil(terms[1] * mass ** terms[0])
 
 
-def metabolic_rate_energy(
```
Comment (dalonsoa) [inline virtual_ecosystem/models/animal/scaling_functions.py:39]: Same here. This function is removed but not added anywhere else and, also, not related with the purpose of the PR. Is it an error or am I missing anything?

**Review by dalonsoa — APPROVED** [review]
I've a couple of questions and suggestions, but the code itself - assuming it is scientifically sound, looks ok.

File: virtual_ecosystem/models/animal/animal_cohorts.py (hunk 3 above)
Comment (TaranRallings) [reply virtual_ecosystem/models/animal/animal_cohorts.py:700]: Both this function and metabolic_rate_energy are no longer in use. They were replaced with inflict_non_predation_mortality and metabolic_rate respectively. I forgot to remove them in the previous relevant PR. I'll try and include that in the PR description if it happens again in the future.

File: virtual_ecosystem/models/animal/animal_cohorts.py (hunk 1 above)
Comment (TaranRallings) [reply virtual_ecosystem/models/animal/animal_cohorts.py:135]: I've going to do a pass through all of my code to rework the TODOs and make issues from them shortly, I'll add this to the list.

**Pull request merged**